lib/utils/report.mjs

```diff
@@ -129,6 +111,42 @@ export default function report(problem) {
 	result.warn(warningMessage, warningProperties);
 }
 
+/**
+ * @param {import('stylelint').Problem} problem
+ * @return {problem is import('stylelint').ProblemWithNormalizedPositions}
+ */
+function normalizeReportSourceLocation(problem) {
+	// Use the given node to calculate the start and end positions.
+	if (problem.node && (!problem.start || !problem.end)) {
+		const range = nodeRangeBy(problem.node, { index: problem.index, endIndex: problem.endIndex });
+
+		if (range) {
+			problem.start ??= range.start;
+			problem.end ??= range.end;
+		}
+	}
+
+	if (!isNumber(problem.line) && isNumber(problem.start?.line)) {
+		// Fallback the line to the line of the start position
+		problem.line = problem.start.line;
+	} else if (!problem.start && isNumber(problem.line)) {
+		// Fallback the start position to the line
+		problem.start = {
+			line: problem.line,
+			column: 1,
+		};
+	}
+
+	delete problem.index;
+	delete problem.endIndex;
```

Comment:
Deleting `index` and `endIndex` ensures that nothing after normalization depends on these values.